### PR TITLE
CH 31 Form Send to RES

### PR DESCRIPTION
**Pull request by tfink419**
## Summary
- This work is behind a feature toggle (flipper): **YES** `veteran_readiness_employment_to_res`
- Redirect ch31 form from VRE service to RES service

## Related issue(s)
- department-of-veterans-affairs/va.gov-team/issues/65386

## Testing done

- [x] *New code is covered by unit tests*
- Have tested in staging/sandbox environments with code in console

## What areas of the site does it impact?
Backend only redirect to new service to handle same form

## Acceptance criteria

- [x]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.
- [x]  Events are being sent to the appropriate logging solution
- [x]  Documentation has been updated (link to documentation)
- [x]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x]  Feature/bug has a monitor built into Datadog or Grafana (if applicable)
- [x]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [x]  I added a screenshot of the developed feature

### CODE REVIEW

File: app/models/saved_claim/veteran_readiness_employment_claim.rb [hunk 1]
```diff
@@ -144,8 +142,20 @@ def upload_to_vbms(doc_type: '1167')
...
+        updated_form['documentId'] = response[:vbms_document_series_ref_id]
+        update!(form: updated_form.to_json)
+      end
     end
+
+    send_vbms_confirmation_email(user)
+  rescue
+    log_message_to_sentry('Error uploading VRE claim to VBMS',
+                          :warn, { uuid: user.uuid })
+    send_to_central_mail!(user)
```
Comment (micahaspyr) [inline app/models/saved_claim/veteran_readiness_employment_claim.rb:158]: Something I missed on cleanup, but we should rename this method to something else, such as 'send_to_benefits_intake' or 'send_to_lighthouse' where it's defined.

File: app/models/saved_claim/veteran_readiness_employment_claim.rb [hunk 2]
```diff
@@ -161,20 +171,28 @@ def send_to_central_mail!(user)
...
     log_message_to_sentry(guid, :warn, { attachment_id: guid }, { team: 'vfs-ebenefits' })
-    @sent_to_cmp = true
-    log_to_statsd('cmp') do
-      process_attachments!
-    end
 
     send_central_mail_confirmation_email(user)
+    @sent_to_cmp = true
+  rescue => e
+    log_message_to_sentry('Error uploading VRE claim to central mail',
```
Comment (micahaspyr) [inline app/models/saved_claim/veteran_readiness_employment_claim.rb:178]: I believe we're still moving away from sentry logging to just use the Rails.logger as it's logs are tied to Datadog

File: lib/res/ch31_form.rb [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+        # addressLine2: client_hash['street2'],
+        # addressLine3: client_hash['street3'],
+        city: client_hash['city'],
+        state: client_hash['state'],
+        postalCode: client_hash['postalCode']
+      }
+    end
+
+    def process_ch_31_error(e, response_body)
+      log_exception_to_sentry(
```
Comment (micahaspyr) [inline lib/res/ch31_form.rb:140]: Rails.logger instead of sentry

**Review by micahaspyr — CHANGES_REQUESTED** [review]

**Review by micahaspyr — APPROVED** [review]

**Review by micahaspyr — DISMISSED** [review]

File: lib/res/ch31_form.rb [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+      # RES/CMSA expects different keys for postal and state for foreign addresses
+      new_address[:internationalPostalCode] = new_address.delete(:zipCode)
+      new_address[:province] = new_address.delete(:stateCode)
+
+      adjusted_new_address
+    end
+
+    def mapped_address_hash(client_hash)
+      {
+        # isForeign: client_hash['country'] != 'USA',
```
Comment (tblackwe) [inline lib/res/ch31_form.rb:119]: several lines commented out, can you explain why/remove/uncomment ?

File: app/models/saved_claim/veteran_readiness_employment_claim.rb [hunk 5]
```diff
@@ -144,41 +142,59 @@ def upload_to_vbms(doc_type: '1167')
...
+    process_attachments!
     @sent_to_cmp = true
-    log_to_statsd('cmp') do
-      process_attachments!
-    end
 
-    send_central_mail_confirmation_email(user)
+    send_lighthouse_confirmation_email(user)
+  rescue => e
+    Rails.logger.error("Error uploading VRE claim to central mail. user uuid: #{user.uuid}. #{e}")
```
Comment (tblackwe) [inline app/models/saved_claim/veteran_readiness_employment_claim.rb:178]: move the dynamic values into logger payload please

File: app/sidekiq/vre/submit1900_job.rb [hunk 6]
```diff
@@ -20,10 +20,10 @@ class Submit1900Job
     def perform(claim_id, user_uuid)
       claim = SavedClaim::VeteranReadinessEmploymentClaim.find claim_id
       user = User.find user_uuid
-      claim.add_claimant_info(user)
       claim.send_to_vre(user)
-    rescue
-      Rails.logger.warn('VRE::Submit1900Job failed, retrying...')
+    rescue => e
+      Rails.logger.warn("VRE::Submit1900Job failed, retrying...: #{e.message}")
```
Comment (tblackwe) [inline app/sidekiq/vre/submit1900_job.rb:25]: move the dynamic values into logger payload please

File: lib/res/ch31_form.rb [hunk 7]
```diff
@@ -0,0 +1,140 @@
...
+      @claim = claim
+    end
+
+    # Submits prepared data derived from VeteranReadinessEmploymentClaim#form
+    #
+    # @return [Hash] the student's address
+    #
+    def submit
+      if @claim.nil?
+        Rails.logger.error("Ch31NilClaimError. user icn: #{@user.icn}")
```
Comment (tblackwe) [inline lib/res/ch31_form.rb:25]: move the dynamic values into logger payload please

File: lib/res/ch31_form.rb (hunk 4 above)
Comment (tfink419) [reply lib/res/ch31_form.rb]: I removed, they are not needed by RES service, I left them commented out until I had determined if they were needed

**Review by micahaspyr — APPROVED** [review]

**Review by ericboehs — APPROVED** [review]

**Pull request merged**